File: src/main/java/demo/Problem1.java
package demo;


import java.util.stream.IntStream;

public class Problem1 {

    public static int getSumOfMultiplesBelow2(int number) {
        return IntStream.iterate(0, n -> n + 1)
                .limit(number)
                .filter( n -> n % 3 == 0 || n % 5 == 0)
                .sum();
    }

    private static boolean multiple(int i, int d) {
        return i % d == 0;
    }

    public static int getSumOfMultiplesBelow1(int number) {

        int sum = 0;

        for (int i = 0; i < number; i++) {
            if (multiple(i, 3) || multiple(i, 5)) {
                sum += i;
            }
        }

        return sum;

    }

}

File: src/main/kotlin/Problem1.kt
package demo

import java.util.*

object KProblem1 {

    fun getSumOfMultiplesBelow2(number: Int): Int {
        return 0.rangeTo(number - 1)
                .filter {multiple(it, 3) || multiple(it, 5)}
                .sum();
    }

    private fun multiple(i: Int, d: Int) = (i % d) == 0;

    fun getSumOfMultiplesBelow1(number: Int): Int {
        var sum = 0;
        for (i in 0 .. number - 1) {
            if (multiple(i, 3) || multiple(i, 5)) {
                sum += i;
            }
        }
        return sum;
    }

}


